client/src/routes/Session/Session.tsx

```diff
@@ -128,23 +127,23 @@ const Session = () => {
   usePreventGoingBack(leaveSessionWithConfirm);
 
   useEffect(() => {
-    if (session?.id) {
+    if (sessionState?.id) {
       logSessionMetricEvent('Enter Sharing Session');
     }
-  }, [logSessionMetricEvent, session?.id]);
+  }, [logSessionMetricEvent, sessionState?.id]);
 
   useEffect(() => {
-    if (session?.exerciseState.completed) {
+    if (sessionState?.completed) {
       logSessionMetricEvent('Complete Sharing Session');
     }
-  }, [session?.exerciseState?.completed, logSessionMetricEvent]);
+  }, [sessionState?.completed, logSessionMetricEvent]);
 
   useEffect(() => {
-    if (session?.ended) {
+    if (sessionState?.ended) {
       leaveMeeting();
```

Comment:
this is async so maybe await it and then navigate. Could return to something unmounted otherwise or perhaps keep it in memory